pkg/resources/role_grants.go

```diff
@@ -0,0 +1,264 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jmoiron/sqlx"
+)
+
+func RoleGrants() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateRoleGrants,
+		Read:   ReadRoleGrants,
+		Delete: DeleteRoleGrants,
+		Update: UpdateRoleGrants,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"role_name": &schema.Schema{
+				Type: schema.TypeString,
+				Elem: &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(strings.ToUpper(v.(string)))
+				},
+				Required:    true,
+				Description: "The name of the role we are granting.",
+				ValidateFunc: func(val interface{}, key string) ([]string, []error) {
+					return snowflake.ValidateIdentifier(val)
+				},
+			},
+			"roles": &schema.Schema{
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(strings.ToUpper(v.(string)))
+				},
+				Optional:    true,
+				Description: "Grants role to this specified role.",
+			},
+			"users": &schema.Schema{
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(strings.ToUpper(v.(string)))
+				},
+				Optional:    true,
+				Description: "Grants role to this specified user.",
+			},
+		},
+		// TODO
+		// Importer: &schema.ResourceImporter{
+		// 	State: schema.ImportStatePassthrough,
+		// },
+	}
+}
+
+func CreateRoleGrants(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := data.Get("name").(string)
+	roleName := data.Get("role_name").(string)
+	roles := expandStringList(data.Get("roles").(*schema.Set).List())
+	users := expandStringList(data.Get("users").(*schema.Set).List())
+
+	if len(roles) == 0 && len(users) == 0 {
+		return fmt.Errorf("No users or roles specified for role grants.")
+	}
+
+	for _, role := range roles {
+		err := grantRoleToRole(db, roleName, role)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, user := range users {
+		err := grantRoleToUser(db, roleName, user)
+		if err != nil {
+			return err
+		}
+	}
+	data.SetId(name)
+	return ReadRoleGrants(data, meta)
+}
+
+func grantRoleToRole(db *sql.DB, role1, role2 string) error {
+	g := snowflake.RoleGrant(role1)
+	_, err := db.Exec(g.Role(role2).Grant())
+	return err
+}
+
+func grantRoleToUsers(db *sql.DB, roleName string, users []string) error {
```

Comment:
// TODO

Reply:
Ha good catch. Not meant to be TODO, not sure why tests didn't catch it.

Reply:
nm actually meant to be deleted

Reply:
:+1: